samcli/lib/remote_invoke/kinesis_invoke_executors.py

```diff
@@ -0,0 +1,132 @@
+"""
+Remote invoke executor implementation for Kinesis streams
+"""
+import logging
+import uuid
+from dataclasses import asdict, dataclass
+from typing import cast
+
+from botocore.exceptions import ClientError, ParamValidationError
+from mypy_boto3_kinesis import KinesisClient
+
+from samcli.lib.remote_invoke.exceptions import (
+    ErrorBotoApiCallException,
+    InvalidResourceBotoParameterException,
+)
+from samcli.lib.remote_invoke.remote_invoke_executors import (
+    BotoActionExecutor,
+    RemoteInvokeIterableResponseType,
+    RemoteInvokeOutputFormat,
+    RemoteInvokeResponse,
+)
+
+LOG = logging.getLogger(__name__)
+STREAM_NAME = "StreamName"
+DATA = "Data"
+PARTITION_KEY = "PartitionKey"
+
+
+@dataclass
+class KinesisStreamPutRecordTextOutput:
+    """
+    Dataclass that stores put_record boto3 API fields used to create
+    text output.
+    """
+
+    ShardId: str
+    SequenceNumber: str
+
```

Comment:
Should these be named as `shard_id` and `sequence_number`?

Reply:
I wanted to name them with an `_` as well but the boto3 API returns the values `SequenceNumber` and `ShardId` and so I thought we should keep the remote invoke output consistent with the API returned response names.